src/viewtransaction/viewTransaction.es6

```diff
@@ -350,7 +350,7 @@ function drawXLines(contract, state, chart) {
     }
 
     if (is_sold_before_expiry) drawXLine({ line_time: sell_time, label: 'end_time', dashStyle: 'Dash' });
-    if (!is_path_dependent) drawXLine({ line_time: date_expiry, label: 'end_time', dashStyle: 'Dash' });
+    if (!is_path_dependent) drawXLine({ line_time: (date_expiry || exit_tick_time), label: 'end_time', dashStyle: 'Dash' });
```

Comment:
I think we always want to use the `exit_tick_time` as basis for end_time, if response doesn't have `exit_tick_time` then better to wait with drawing the end time until `exit_tick_time` is received

Reply:
`drawXLines` is always executed when the tick comes. for days contract, we need to use date_expiry as end time

Reply:
It's drawn based on the `proposal_open_contract` stream / response, I think days also have exit_tick_time?

Reply:
yes, they have exit tick time, but, i dont think we can draw exit tick time for end time except path dependent contract. bcause in binary static, its totally the date expiry. i think i have to find a way. to make it draw the date_expiry regardless it exist or not.

Reply:
one solution might be triggering to draw chart after the contract finish